src/coreclr/ilasm/assem.cpp

```diff
@@ -315,6 +315,25 @@ BOOL Assembler::AddMethod(Method *pMethod)
     {
         if(fIsImport || IsMdAbstract(pMethod->m_Attr) || IsMdPinvokeImpl(pMethod->m_Attr)
            || IsMiRuntime(pMethod->m_wImplAttr) || IsMiInternalCall(pMethod->m_wImplAttr)) return TRUE;
+
+        // Check method attributes that indicate no method body is expected.
```

Comment:
I think it would be better to delete this sanity check from ilasm instead. ilasm should not be involved in figuring out when it is ok to have no method body.

Reply:
That is fine, but then we need to figure out a way to ensure the IL round tripping tests work. I am loathe to disable them because special casing these operations creates odd support holes and surprises in future scenarios.

I see two options:
1) Leave this support in ILAsm.
2) Delete the code in ILAsm that injects a `ret` into the IL stream.

The second would seem to follow the spirit of what I think you expect from ILAsm - assemble what the user has written and don't be "helpful".

Reply:
Yes, the second is what I meant.